src/main/java/org/folio/inventory/validation/experimental/AbstractTargetItemStatusValidator.java

```diff
@@ -0,0 +1,44 @@
+package org.folio.inventory.validation.experimental;
+
+import org.folio.inventory.domain.items.Item;
+import org.folio.inventory.domain.items.ItemStatusName;
+import org.folio.inventory.exceptions.UnprocessableEntityException;
+import org.folio.inventory.support.http.server.ValidationError;
+
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+
+import static org.folio.inventory.support.CompletableFutures.failedFuture;
+
+public abstract class AbstractTargetItemStatusValidator {
+  private ItemStatusName itemStatusName;
+  private Set<ItemStatusName> allowedStatusToMark;
+
+  protected AbstractTargetItemStatusValidator(ItemStatusName itemStatusName, Set<ItemStatusName> allowedStatusToMark) {
+    this.itemStatusName = itemStatusName;
+    this.allowedStatusToMark = allowedStatusToMark;
+  }
+
+  public CompletableFuture<Item> itemHasAllowedStatusToMark(Item item) {
```

Comment:
To me, this method refuses items from being marked when they are in an invalid source status. I suggest we change the name to reflect that intent.

```suggestion
  public CompletableFuture<Item> refuseItemWhenNotInAcceptableSourceStatus(Item item) {
```